src/util/prop.ts

```diff
@@ -44,11 +44,30 @@ export const makePropWatcher = <T, E>(
  * @param {Ref<E>} target - A Ref representing the target object to be updated.
  * @param {() => E & { dispose?(): void }} newPlainObjectFunction - A function that creates a new plain object to retrieve the defaults from with an optional "dispose" method for cleanup.
  */
-export const makePropWatchers = <T, E>(
-  propGettersAndPropertyPaths: (string | (() => T))[][],
+export const makePropWatchers = <E>(
```

Comment:
I find this really useful, @andretchen0 @JaimeTorrealba maybe we should consider it cross-packages.

Do we have enough utils around to make a package?